### ISSUE TITLE
Install completely broken

### ISSUE THREAD

**Issue by joshribakoff** [prisma/prisma#18100]
### Bug description

Your "generate" command is trying to force me to put `prisma` as a dev dependency. That is wrong, it is a dependency. Nevertheless, it doesn't work. If you move it to dev dependencies, you get an inverse/circular error telling you to change it back to a regular dependency.

### How to reproduce

Just `yarn add prisma` and run `yarn prisma generate`:

```
➜  git:(main) ✗ yarn prisma generate
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Usage Error: Package "prisma" is already listed as a regular dependency - remove the -D,-P flags or remove it from your dependencies first

$ yarn add [--json] [-E,--exact] [-T,--tilde] [-C,--caret] [-D,--dev] [-P,--peer] [-O,--optional] [--prefer-dev] [-i,--interactive] [--cached] [--mode #0] ...
Error: Command failed with exit code 1: yarn add -D prisma@4.10.1


Error: Could not resolve @prisma/client despite the installation that we just tried.
Please try to install it by hand with yarn add @prisma/client and rerun yarn prisma generate 🙏.


➜ git:(main) ✗ yarn remove prisma
➜  git:(main) ✗ yarn add -D prisma  
➜  git:(main) ✗ yarn prisma generate
Error: Could not resolve @prisma/client despite the installation that we just tried.
Please try to install it by hand with yarn add @prisma/client and rerun yarn prisma generate 🙏.
```

### Expected behavior

Generate the client, do not mess with me dependencies

### Prisma information

n/a

### Environment & setup

- OS: macOS,
- Database:  n/a
- Node.js version: 18.14.1

### Prisma Version

```
4.10.1
```

**Comment by joshribakoff** [comment]
By the way, if it is your claim I'm using an unsupported setup or something, I would in turn ask you why your error message is so unhelpful.

**Comment by joshribakoff** [comment]
Yep, of course switching from pnp to node-modules fixed it, but I am using pnpify and custom prisma output locations in other projects. Because of this issue, I cannot replicate my previously working setup, even w/ the same versions of prisma, yarn,  etc. :/

Eventually I solved it - had to run `yarn pnpify prisma generate` instead of `yarn prisma generate`.

Perhaps this issue could be renamed "please add documentation for pnpify workarounds" or "improve error messages when users try to use pnp in unsupported manner"

**Comment by cybermelons** [comment]
> Eventually I solved it - had to run `yarn pnpify prisma generate` instead of `yarn prisma generate`.

This worked for me! I'm using yarn 3.5.0

**Comment by zach-betz-hln** [comment]
Related: https://github.com/prisma/prisma/issues/8765#issuecomment-1854451597
